### ISSUE TITLE
Add built-in cross-validation sampling

### ISSUE THREAD

**Issue by michellewang** [fedbiomed/fedbiomed#1616]
It would be very nice if it were possible to do CV with Fed-BioMed. Currently the common practice seems to be to subset the data into different folds/splits *prior* to running Fed-BioMed. This is suboptimal since it creates duplication of data and extra work since each training split needs to be added to nodes as separate datasets.

I think something like this could make sense:

1. The researcher can specify CV args when creating the Fed-BioMed `Experiment`
2. The training plan has an extra optional method that takes as input a `Dataset` and returns sample indices to use. By default this method would just return all the indices (i.e. no CV)
3. Fed-BioMed subsets the data as appropriate before `__getitem__` is called

Example training plan:
```python
import numpy as np
from fedbiomed.common.dataset import Dataset
from sklearn.model_selection import KFold

class MyTrainingPlan(...):

    # basic k-fold
    def cross_validation_sampling(
        self,
        dataset: Dataset,
        i_split: int = 0,
        n_splits: int = 5,
        shuffle: bool = True,
        random_state: int = None,
    ):
        cv = KFold(n_splits=n_splits, shuffle=shuffle, random_state=random_state)
        splits = cv.split(np.arange(len(dataset)))
        for _ in range(i_split + 1):
            idx_train, _ = next(splits)
        return idx_train  # ndarray
```

And the experiment construction would look like this:
```python
from fedbiomed.researcher.experiment import Experiment

experiment = Experiment(
    training_plan=MyTrainingPlan,
    cv_args={  # passed to the training plan's cross_validation_sampling()
        "i_split": 0,
        "n_splits": 10,
        "shuffle": True,
        "random_state": 12345,
    },
    # ...
)
```

I think something like this would allow maximal flexibility, which would be needed with more complex/different cases, for example sklearn's `StratifiedKFold`/`GroupKFold`/`StratifiedGroupKFold`/`ShuffleSplit`/etc.

Additional thoughts:
1. Ideally the indices are saved
2. It could be useful to also save the test indices, which in my example are discarded
3. I don't know if Fed-BioMed should compute the test scores or no. In my analysis I do that separately after having run Fed-BioMed, which is probably the easiest way to go, though that would require `cross_validation_sampling` to also return test indices

Happy to discuss ideas further!
